python/lsst/verify/ap/metrics.py

```diff
@@ -0,0 +1,157 @@
+    using the `parent` parameter.
+    """
+
+    def __init__(self):
+        # Help and documentation will be handled by main program's parser
+        argparse.ArgumentParser.__init__(self, add_help=False)
+        self.add_argument('--silent', dest='submit_metrics', action='store_false',
+                          help='Do NOT submit metrics to SQuaSH (not yet implemented).')
+        # Config info we don't want on the command line
+        self.set_defaults(user=os.getenv(_ENV_USER), password=os.getenv(_ENV_PASSWORD),
+                          squash_url=os.getenv(_ENV_URL, _SQUASH_DEFAULT_URL))
+
+
+class AutoJob:
+    """A wrapper for an lsst.verify.Job that automatically handles
+    initialization and shutdown.
+
+    When used in a `with... as...` statement, the wrapper assigns the
+    underlying job to the `as` target.
+
+    This object shall always attempt to dump metrics to disk, but shall only
+    submit to SQuaSH if the program ran without errors.
+
+    Parameters
+    ----------
+    parsed_cmd_line: argparse.Namespace
+        Command-line arguments, including all arguments supported by `MetricsParser`.
+    """
+    def __init__(self, args):
+        self._job = lsst.verify.Job.load_metrics_package()
+        # TODO: add Job metadata (camera, filter, etc.) in DM-11321
+        self._submit_metrics = args.submit_metrics
+        self._squash_user = args.user
+        self._squash_password = args.password
+        self._squash_url = args.squash_url
+
+    def _save_measurements(self, fileName):
+        """Save a set of measurements for later use.
+
+        Parameters
+        ----------
+        fileName:
+            The file to which the measurements will be saved.
+        """
+        self.job.write(fileName)
+
+    def _send_to_squash(self):
+        """Submits a set of measurements to the SQuaSH system.
+
+        Parameters
+        ----------
+        fileName: string
+            a file containing measurements in lsst.verify format
+        """
+        self.job.dispatch(api_user=self._squash_user, api_password=self._squash_password,
+                          api_url=self._squash_url)
+
+    @property
+    def job(self):
+        """The Job contained by this object.
+        """
+        return self._job
+
+    def __enter__(self):
+        """Allow the underlying Job to be used in with statements.
+        """
+        return self.job
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        """Packages all metric measurements performed during this run.
+
+        The measurements shall be exported to `verify_ap.verify.json`, and the
+        metrics framework shall be shut down. If the context was exited
+        normally and the appropriate flag was passed to this object's
+        constructor, the measurements shall be sent to SQuaSH.
+        """
+        log = lsst.log.Log.getLogger('verify.ap.metrics.AutoJob.__exit__')
+
+        out_file = 'verify_ap.verify.json'
```

Comment:
It's the default name assumed by `verify`, according to SQR-019, and there doesn't seem an obvious reason to let the user create multiple dump files.